cpp/cgi_progs/newsletter_uploader.py

```diff
@@ -1,30 +1,42 @@
 #!/usr/bin/env python
 import cgi
 import cgitb
+import datetime
 import os
 
 
+print("Content-Type: text/html\n")
 cgitb.enable(display=0, logdir="/usr/local/var/log/tuefind/")
-form = cgi.FieldStorage()
-
-
-# A nested FieldStorage instance holds the file
-file_field = form['file']
 
 
 # Test if the file was uploaded
-if file_field:
-    # Strip leading path from file name to avoid directory traversal attacks
-    filename = os.path.basename(file_field.filename)
-    open("/usr/local/var/lib/tuelib/newsletters/" + filename, 'w').write(file_field.file.read())
-    message = 'The file "' + filename + '" was uploaded successfully.'
+form = cgi.FieldStorage()
+if "file" not in form:
+    # Show upload form
+    message = """
+        The uploaded file must be a plain-text file.  The first line will be used as the subject of the
+        email message and the remainder as the message body.  The encoding of the file must be UTF-8 with
+        no leading BOM.
```

Comment:
Let's add "(byte-order marker)" in parentheses because we're usually dealing with muggles.